asserts/crypto.go

```diff
@@ -290,14 +303,41 @@ func OpenPGPPrivateKey(privk *packet.PrivateKey) PrivateKey {
 	return openpgpPrivateKey{privk}
 }
 
-func generatePrivateKey() (*packet.PrivateKey, error) {
+// GenerateKey generates a private/public key pair.
+func GenerateKey() (PrivateKey, error) {
 	priv, err := rsa.GenerateKey(rand.Reader, 4096)
 	if err != nil {
 		return nil, err
 	}
-	return packet.NewRSAPrivateKey(time.Now(), priv), nil
+	return OpenPGPPrivateKey(packet.NewRSAPrivateKey(time.Now(), priv)), nil
 }
 
 func encodePrivateKey(privKey PrivateKey) ([]byte, error) {
 	return encodeKey(privKey, "private key")
 }
+
+type sealedPrivateKey struct {
+	pubk   *packet.PublicKey
+	signer func([]byte, *packet.Config) (*packet.Signature, error)
+}
+
+func (spk *sealedPrivateKey) PublicKey() PublicKey {
+	return OpenPGPPublicKey(spk.pubk)
+}
+
+func (spk *sealedPrivateKey) keyEncode(w io.Writer) error {
+	return fmt.Errorf("cannot encode a sealed private key")
+}
+
+func (spk *sealedPrivateKey) keyFormat() string {
+	return ""
+}
+
+func (spk *sealedPrivateKey) sign(content []byte, cfg *packet.Config) (*packet.Signature, error) {
+	return spk.signer(content, cfg)
+}
+
+// SealedOpenPGPPrivateKey returns a PrivateKey for database use realized as pair of a opengpg packet.PublicKey and a signer function for the sealed private key part.
+func SealedOpenPGPPrivateKey(pubKey *packet.PublicKey, signer func([]byte, *packet.Config) (*packet.Signature, error)) PrivateKey {
```

Comment:
This feels like a strange interface. Would appreciate a bit more insight into the design here.